### PR TITLE
🚀 Improve error handling for net error(s)

### PR DESCRIPTION
**Pull request by ReneWerner87**
Fixes: reverse proxy support #2419

With this change we want to improve the default error handler, so that it does not give out the internal network information of the dial in case of net.error.
So that fiber becomes more secure.

### CODE REVIEW

**Review by leonklingele — COMMENTED** [review]
Is it really always an `502 Bad Gateway` error?

**Comment by gaby** [comment]
> Is it really always an `502 Bad Gateway` error?

I'm wondering the same thing 🤔

**Comment by ReneWerner87** [comment]
just give me good suggestions and i will change it

before it was always a 500 that was not specific and exposed the error message with the network information.

so i would either take the 500 again or 502 or 503

**Comment by gaby** [comment]
> just give me good suggestions and i will change it
> 
> before it was always a 500 that was not specific and exposed the error message with the network information.
> 
> so i would either take the 500 again or 502 or 503

Is it possible to get the return code from the NetError object?

**Comment by ReneWerner87** [comment]
not really, because it is not a http error
its a network connection error when trying the client request internally

**Comment by gaby** [comment]
@ReneWerner87 This is what GPT-4 says:

```
This function appears to be a default error handler for some web application, and it is handling errors and returning appropriate HTTP status codes. Given the provided code, it makes sense to return a 500 status code (Internal Server Error) by default.

The function first checks if the error is a net.Error, and if it is, it sets the error to ErrBadGateway. This would typically correspond to a 502 status code (Bad Gateway). Then, the function checks if the error is of type *Error, and if it is, it retrieves the status code from that error.

However, it is worth noting that the function does not explicitly return 503 (Service Unavailable) at any point. To include that, you might want to add another condition to check for specific errors corresponding to a 503 status code.

In summary, based on the provided code, it makes sense for the function to return 500 (Internal Server Error) by default, and 502 (Bad Gateway) if the error is a net.Error. To return a 503 (Service Unavailable), you would need to add additional logic to handle that case.
```

I guess bad gateway makes sense in terms of a reverse proxy.

**Comment by ReneWerner87** [comment]
- [ ] add an unittest
